### PR TITLE
feat: added gitlab pipelines provider

### LINKED ISSUE #326: Add Gitlab Pipelines Provider
Users should be able to `notify` (action) Gitlab Pipelines, `GitlabpipelinesProvider` should be implemented with the necessary methods implemented.

The provider should be able to execute a [GitLab Pipeline](https://docs.gitlab.com/ee/api/pipelines.html) with parameters.

### PR DESCRIPTION
**Pull request by melsonic**
Closes #326

### Description
Added Gitlab Pipelines Provider to interact with the Gitlab Pipelines API.

/claim #326

### CODE REVIEW

**Comment by talboren** [comment]
Thanks! Will review it ASAP :)

Can you take a look at https://docs.keephq.dev/ and add the provider documentation? (How to get the webhook url, etc.)

Please also add a PNG of GitLab pipelines icon to keep-up/public/icons 🙏🏼

**Comment by melsonic** [comment]
> Thanks! Will review it ASAP :)
> 
> Can you take a look at https://docs.keephq.dev/ and add the provider documentation? (How to get the webhook url, etc.)
> 
> Please also add a PNG of GitLab pipelines icon to keep-up/public/icons 🙏🏼

Sure, I will update the documentation and the icon, and if the review recommends some change I will do that altogether.

**Review by shahargl — CHANGES_REQUESTED** [review]
lgtm just few small things - 
1. add the svg
2. add docs (under docs/) how to use/get the token

**Comment by melsonic** [comment]
> lgtm just few small things -
> 
> 1. add the svg
> 2. add docs (under docs/) how to use/get the token

Sure, I will add these changes ASAP :+1:

**Comment by melsonic** [comment]
> lgtm just few small things -
> 
> 1. add the svg
> 2. add docs (under docs/) how to use/get the token

Hi @shahargl before pushing a commit, please clarify these two points
1. Is this the correct logo for Gitlab Pipelines? 
IMG_20230926_173331
2. Is there any preferable format for the provider documentation?

**Comment by melsonic** [comment]
please review the changes @shahargl

**Review by shahargl — APPROVED** [review]
lgtm

File: keep/providers/gitlabpipelines_provider/gitlabpipelines_provider.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+    """
+    access_token: str = dataclasses.field(
+        metadata={
+            "required": True,
+            "description": "GitLab Access Token",
+            "sensitive": True,
+        }
+    )
+
+class GitlabPipelinesProvider(BaseProvider):
```
Comment (shahargl) [inline keep/providers/gitlabpipelines_provider/gitlabpipelines_provider.py:27]: This should be GitlabpipelinesProvider

**Review by shahargl — CHANGES_REQUESTED** [review]
change the class name to GitlabpipelinesProvider so tests will pass

File: keep/providers/gitlabpipelines_provider/gitlabpipelines_provider.py (hunk 1 above)
Comment (melsonic) [reply keep/providers/gitlabpipelines_provider/gitlabpipelines_provider.py:27]: sure

**Comment by melsonic** [comment]
Hi, @shahargl how can I claim this bounty? it is not showing in the dashboard

**Pull request merged**